OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java

```diff
@@ -827,15 +829,15 @@ public String getHighway() {
 		return getHighway(types, region);
 	}
 
-	public boolean hasPrivateAccess() {
+	public boolean hasPrivateAccess(GeneralRouterProfile profile) {
 		int sz = types.length;
 		for (int i = 0; i < sz; i++) {
-			RouteTypeRule r = region.quickGetEncodingRule(types[i]);
-			if ("motorcar".equals(r.getTag())
-					|| "motor_vehicle".equals(r.getTag())
-					|| "vehicle".equals(r.getTag())
-					|| "access".equals(r.getTag())) {
-				if (r.getValue().equals("private")) {
+			RouteTypeRule rule = region.quickGetEncodingRule(types[i]);
+			String tag = rule.getTag();
+			if ("vehicle".equals(tag) || "access".equals(tag)
```

Comment:
Faster to switch if's
```if (rule.getValue().equals("private")) {
 String tag = rule.getTag();
			if ("vehicle".equals(tag) || "access".equals(tag)
```

Reply:
Code is hard to read...
Probably it's time to split by many if's and group by profiles

Reply:
split by profiles